### PR TITLE
DOC: Rename the FDA SW development guidelines doc file

### PR DESCRIPTION
**Pull request by jhlegarreta**
Rename the FDA software development guidelines documentation file, and change the corresponding cross-reference in the FAQ document.

Take advantage of the commit to rename the headings to match the original Wiki contents, and use sentence style case.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Documentation/docs/learn/fda_sw_development_guidelines.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# FDA Guidelines for Off-the-Shelf software
+# FDA guidelines for software development
+
+## FDA guidelines for off-the-shelf software
```
Comment (N-Dekker) [inline Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Thanks Jon. While I agree about using sentence style case for section headers, I would prefer _title case_ for the title. Just like in the original wiki page (https://itk.org/Wiki/FDA_Guidelines_for_Software_Development):

    FDA Guidelines for Software Development

File: Documentation/docs/learn/faq.md [hunk 2]
```diff
@@ -346,7 +346,7 @@ be summarized as follows:
  guidelines for the development of software-related medical devices."
 ```
 
-For mode details see the page [FDA Guidelines for Software Developement](fda_off_the_shelf_sw_guidelines.md).
+For mode details see the page [FDA Guidelines for Software Developement](fda_sw_development_guidelines.md).
```
Comment (N-Dekker) [inline Documentation/docs/learn/faq.md:349]: Can you possibly still fix the typo in "Developement" with this pull request?

Comment (jhlegarreta) [reply Documentation/docs/learn/faq.md]: Done in bc13560.

File: Documentation/docs/learn/fda_sw_development_guidelines.md (hunk 1 above)
Comment (jhlegarreta) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Sorry Niels, but I am being as consistent as possible with the rest of the headings.

Comment (N-Dekker) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Thanks for your reply, but sorry, but I still prefer title case for titles. The _Python Quick Start_ also uses title case:
 
https://github.com/InsightSoftwareConsortium/ITK/blob/4543ba6dce61bdc4540bb0641b2aafae10056b9a/Documentation/docs/learn/python_quick_start.md?plain=1#L1

So why would we not using title case for titles, consistently?

Comment (jhlegarreta) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Niels, I did not contribute to that file, and I said that in the files I contributed I was being as consistent _as possible_. Please do change all of them in a separate PR if you feel like they should be, but this PR does address your original request.

Comment (N-Dekker) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Thanks for addressing my request, Jon. I understand now, you just want to be consistent with your previous contributions. So don't you _like_ title case at all?

Anyway, it's no big deal, I won't press the "Request changes" button.

Comment (jhlegarreta) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: > So don't you like title case at all?

No, I don't. But as said, feel free to open a PR that changes them all.

Comment (N-Dekker) [reply Documentation/docs/learn/fda_sw_development_guidelines.md:3]: Thanks for your clarification!

**Review by thewtex — APPROVED** [review]
These iterative improvements are how we progress!

**Pull request merged**